NewsApp/Controllers/NewsViewController/Cell/CountrySelectionView.swift

```diff
@@ -0,0 +1,98 @@
+        return imageView
+    }()
+    
+    private lazy var countryCodeLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont.systemFont(ofSize: 16, weight: .medium)
+        label.textColor = .black
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    private lazy var arrowImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(systemName: "chevron.down")
+        imageView.tintColor = .black
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.isUserInteractionEnabled = true
+        setupUI()
+        setupTapGesture()  // Ensure tap gesture is set up to trigger the closure
+        print("CountrySelectionView Frame: \(self.frame)")
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupUI() {
+        addSubview(flagImageView)
+        addSubview(countryCodeLabel)
+        addSubview(arrowImageView)
+        
+        
+        flagImageView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        flagImageView.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+        flagImageView.widthAnchor.constraint(equalToConstant: 24).isActive = true
+        flagImageView.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        
+        countryCodeLabel.leadingAnchor.constraint(equalTo: flagImageView.trailingAnchor, constant: 8).isActive = true
+        countryCodeLabel.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+        
+        arrowImageView.leadingAnchor.constraint(equalTo: countryCodeLabel.trailingAnchor, constant: 8).isActive = true
+        arrowImageView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        arrowImageView.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+        arrowImageView.widthAnchor.constraint(equalToConstant: 12).isActive = true
+        arrowImageView.heightAnchor.constraint(equalToConstant: 12).isActive = true
+        
+    }
+    
+    private func setupTapGesture() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap))
+        self.addGestureRecognizer(tapGesture)
+    }
+    
+    @objc private func handleTap() {
+        print("CountrySelectionView tapped")  // Debugging print statement
+        onTap?()
+    }
+    
+    func configure(flag: UIImage?, countryCode: String) {
+        flagImageView.image = flag
+        countryCodeLabel.text = countryCode
+    }
+}
+
+extension CountrySelectionView: UIGestureRecognizerDelegate {
+    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+}
```

Comment:
What does this extension do? I am asking because I have never seen it before when implementing gestureReconisers.

Reply:
here I had a problem with my containerView listening to tap gesture, it wasn't responding so the internet suggested that the extension because it is useful when you need more control over how gestures are recognized, especially in complex interfaces where multiple gestures might interact with the same view.